### DISCUSSION TITLE
How to change font weight of text

### DISCUSSION THREAD

**Discussion by oTnTh** [orbitalquark/textadept#189] [Q&A] [answered]
I tried this in init.lua

```
events.connect(events.LEXER_LOADED, function(name)
	if name == 'text' then
		view:set_theme('dark', {font = 'Source Han Sans SC', size = 14, weight = 500})
	end
end)
```
"weight = 500" didn't work, and the default font weight is too thin for CJK chars.

Thanks.

**Comment by oTnTh** [comment]
Also, can I specify a fallbak font, like CSS?

```
font-family: Consolas, "Source Han Sans SC"
```

**Comment by orbitalquark** [comment]
Your font has to support multiple weights. I don't know if that font does. Maybe just try bold?

You cannot specify fallback fonts like in CSS. The font you choose has to exist.

**Comment by oTnTh** [comment]
<img width="369" alt="fontselect" src="https://user-images.githubusercontent.com/92753/158993714-32d32da0-efd9-4342-9510-d3d5ad95f918.png">

https://github.com/adobe-fonts/source-han-sans

Didn't work:

```
view:set_theme('dark', {font = 'Source Han Sans SC', size = 14, weight = 700})
view:set_theme('dark', {font = 'Source Han Sans SC Bold', size = 14})
view:set_theme('dark', {font = 'Source Han Sans SC', size = 14, bold = true})
```

Win10, textadept_nightly_2022-03-17.win

Any idea? Thanks.

**Comment by orbitalquark** [comment, marked as answer]
Sorry, I misread your original message. Textadept's default themes do not allow setting weight via `view:set_theme()`. You will need to modify the actual theme[1]. You can either do this:

  styles.default = {font = font, size = size, fore = colors.light_black, back = colors.white, weight = 700}

Or this:

  styles.default = {font = font, size = size, fore = colors.light_black, back = colors.white, weight = weight}

and keep your `view:set_theme('light', {..., weight = 700})`.

Try that. Hopefully you'll get better results.

[1]: https://orbitalquark.github.io/textadept/manual.html#themes

**Comment by oTnTh** [comment]
Thanks for you help, dark.lua did the work for me.

"monospace" isn't a real font, so I think it's might be a way to specify a fallback font.

I try to add this to `textadept\etc\fonts\fonts.conf`:

```
<alias>
	<family>monospace</family>
	<prefer>
		<family>Consolas</family>
		<family>Source Han Sans SC</family>
	</prefer>
</alias>
```

Unfortunately, didn't work.
